cmake/dependencies.cmake

```diff
@@ -0,0 +1,109 @@
+# Download all external projects in the specified directory.
+include(ExternalProject)
+set(EP_PREFIX "${PROJECT_SOURCE_DIR}/external")
+set_directory_properties(PROPERTIES
+  EP_PREFIX ${EP_PREFIX}
+)
+
+include(cmake/DownloadProject.cmake)
+
+# Project dependencies.
+find_package(Threads REQUIRED)
+
+################################
+#         flatbuffers          #
+################################
+download_project(PROJ flatbuffers
```

Comment:
Versioning has to be supported by find_packages in this case. For example, [FindFlatBuffers.cmake](https://github.com/google/flatbuffers/blob/master/CMake/FindFlatBuffers.cmake) does not seem to support this feature.